ams/server/src/main/java/com/netease/arctic/server/optimizing/plan/AbstractPartitionPlan.java

```diff
@@ -316,6 +316,11 @@ public List<SplitTask> splitTasks(int targetTaskCount) {
                   rewritePosDataFiles.add(f.getFile());
                   deleteFiles.addAll(f.getDeleteFiles());
                 });
+        if (rewriteDataFiles.size() == 1
```

Comment:
Skipping files during task splitting will result in data loss for Mixed Hive Tables, because Mixed Hive tables requires all files within a partition must be involved in an optimizing process.

Reply:
Maybe we can control whether to filter through a parameter?

Reply:
Yes, it may be a feasible approach, but I don't like quietly filtering out some files during splitting tasks, which goes against intuition. Perhaps we should exclude the unnecessary tasks after the split. 

You may introduce an "afterSplit" operation to perform this, similar to `com.netease.arctic.server.optimizing.plan.AbstractPartitionPlan#beforeSplit`.